core/src/main/java/org/apache/carbondata/core/datamap/TableDataMap.java

```diff
@@ -499,4 +501,26 @@ public DataMapFactory getDataMapFactory() {
     }
     return prunedSegments;
   }
+
+  /**
+   * Prune the datamap of the given segments and return the Map of blocklet path and row count
+   *
+   * @param segments
+   * @param partitions
+   * @return
+   * @throws IOException
+   */
+  public Map<String, Integer> pruneBlockRowCount(List<Segment> segments,
+      final List<PartitionSpec> partitions) throws IOException {
+    Map<String, Integer> blockletToRowCountMap = new HashMap<>();
```

Comment:
what is the usage of partitions?  this method doesn't use it

Reply:
Please check the code below, partitions are used